### PR TITLE
Adding parse results to specstorage

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding parse results to spec storage

Enhancing Parsing Results for Spec Storage

We're integrating functionality to parse XML results and store them in a CSV file. Additionally, we're introducing the ability to include extra parameters based on the run and configurations, which will be essential for the database population. These extra parameters could encompass details such as the Ceph version and MDS configurations.

The CSV file generated will serve as a versatile source, enabling easy loading into any database using database adaptors. This approach facilitates seamless data integration and management, providing flexibility for diverse database environments

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-V919MN/

CSV file : https://docs.google.com/spreadsheets/d/1AIQzuEVupJ-0FCkgK4-jvZqsqfXrKthAvhpfTmsPUyg/edit#gid=0

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
@AmarnatReddy As discussed, it will be a good addition to have the benchmark type mentioned in the csv report generated.

**Comment by AmarnatReddy** [comment]
> @AmarnatReddy As discussed, it will be a good addition to have the benchmark type mentioned in the csv report generated.

Hi Pranav,
Updated to include benchmark field and same logs have been attached

Regards,
Amarnath

**Review by neha-gangadhar — APPROVED** [review]
Approved

File: cli/io/spec_storage.py [hunk 1]
```diff
@@ -159,3 +161,39 @@ def run_spec_storage(
...
+        tree = ET.parse(remote_file)
+        root = tree.getroot()
+
+        metrics = kwargs
+        business_metric = root.find(".//business_metric").text
+        metrics["business_metric"] = business_metric
+        benchmark = root.find(".//benchmark").attrib["name"]
+        metrics["benchmark"] = benchmark
+        for metric in root.findall(".//metric"):
+            name = metric.attrib["name"]
```
Comment (sumabai) [inline cli/io/spec_storage.py:186]: Can we also include units with metric name like for,

-  throughput - KB/s or MB/s or Gbps as applicable
-  'secs' for metrics such as latency
- KB/MB/GB for size

**Pull request merged**